service/src/main/java/bio/terra/cbas/initialization/CloneRecoveryService.java

```diff
@@ -39,58 +41,77 @@ public void cloneRecovery() {
         cbasContextConfig.getWorkspaceId(),
         cbasContextConfig.getWorkspaceCreatedDate());
 
-    pruneCloneSourceWorkspaceHistory();
-  }
+    List<Method> clonedMethods =
+        methodDao.getMethods().stream().filter(this::isMethodCloned).toList();
 
-  @WriteTransaction
-  public void pruneCloneSourceWorkspaceHistory() {
-    List<Method> methods = methodDao.getMethods().stream().filter(this::isMethodCloned).toList();
+    List<MethodTemplateUpdateManifest> templateUpdateManifests =
+        clonedMethods.stream()
+            .map(m -> runSetDao.getRunSetsWithMethodId(m.methodId()))
+            .map(this::generateTemplateUpdateManifest)
+            .toList();
 
-    methods.forEach(this::updateMethodTemplate);
+    templateUpdateManifests.stream()
+        .map(MethodTemplateUpdateManifest::templateRunSets)
+        .forEach(rsList -> rsList.forEach(rs -> runSetDao.updateIsTemplate(rs.runSetId(), true)));
 
-    methods.forEach(this::cleanupClonedMethodHistory);
-  }
+    templateUpdateManifests.stream()
+        .map(MethodTemplateUpdateManifest::nonTemplateRunSets)
+        .forEach(rsList -> rsList.forEach(rs -> runSetDao.updateIsTemplate(rs.runSetId(), false)));
 
-  public Boolean isMethodCloned(Method m) {
-    return !m.originalWorkspaceId().equals(cbasContextConfig.getWorkspaceId());
+    clonedMethods.forEach(
+        method -> {
+          List<RunSet> methodRunSets = runSetDao.getRunSetsWithMethodId(method.methodId());
```

Comment:
This looks like a duplicate call to line 49. Perhaps we can call this once and save the result?

Reply:
it is a duplicate, but the run sets change between line 49 and this one. I could maintain a list of run sets and update their fields as I call methods like `updateIsTemplate`, but this seemed simpler